pkg/reconciler/openshift/openshift_test.go

```diff
@@ -0,0 +1,81 @@
+package openshift
+
+import (
+	"testing"
+
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"gotest.tools/assert"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+func TestReconcileArgoCD_reconcileApplicableClusterRole(t *testing.T) {
+
+	logf.SetLogger(logf.ZapLogger(true))
+	a := makeTestArgoCDForClusterConfig()
+	testClusterRole := &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      a.Name + "-" + testApplicationController,
+			Namespace: a.Namespace,
+		},
+		Rules: makeTestPolicyRules(),
+	}
+	if err := reconcilerHook(a, testClusterRole); err != nil {
```

Comment:
I'm suspect this should use `assert.NilError` (there are a number of places where this could be used).